jmclient/jmclient/taker.py

```diff
@@ -740,13 +766,7 @@ def self_sign(self):
             script = self.wallet.addr_to_script(self.input_utxos[utxo]['address'])
             amount = self.input_utxos[utxo]['value']
             our_inputs[index] = (script, amount)
-
-        # FIXME: ugly hack
-        tx_bin = btc.deserialize(unhexlify(btc.serialize(self.latest_tx)))
-        self.wallet.sign_tx(tx_bin, our_inputs)
-
-        self.latest_tx = btc.deserialize(hexlify(btc.serialize(tx_bin)).decode('ascii'))
-
+        self.latest_tx = self.wallet.sign_tx(self.latest_tx, our_inputs)
```

Comment:
Does this work without the hexlify/unhexlify hack? We wouldn't want to accidentally end up with hex and non-hex data in the tx object.

Reply:
As noted in the earlier comment [here](https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/268#discussion_r244640376), because of the behaviour (now correctly documented) of jmbitcoin.sign, this will always come out hex encoded.